### DISCUSSION TITLE
RocketMQ5.1.3版本，为什么controller模式2主1从的性能比controller模式1主1从的性能低？

### DISCUSSION THREAD

**Discussion by onejimmyboy** [apache/rocketmq#7357] [Q&A]
1：版本 RocketMQ5.1.3
2：jdk1.8
3：集群形态：DLedger Controller 模式
集群1：嵌入namesrv，3个节点（1个controller，2个broker+controller），1主1从
集群2：嵌入namesrv，5个节点（1个controller，2个broker+controller，2个broker），2主1从
4：现象
在使用./producer.sh测试生成tps时，发现1主1从的tps要明显高于2主1从的tps，那按照这种现象，如果多个主的话tps会很低的，岂不是性能很低。面对大数据量的情况下，是不是扩节点就会影响tps性能了？
![image](https://github.com/apache/rocketmq/assets/57432151/b6cb8595-f49c-4514-a2ef-78fc92ed2fbd)
![image](https://github.com/apache/rocketmq/assets/57432151/10337047-ee00-4083-bbe0-3bf729a1127e)

**Comment by onejimmyboy** [comment]
这正常吗？

**Comment by RongtongJin** [comment]
怎么会有2主1从，是不是1主2从，还是说是1主备 + 一个单主？

**Comment by onejimmyboy** [comment]
> 怎么会有2主1从，是不是1主2从，还是说是1主备 + 一个单主？
1主备+1主备，应该叫2主2从，同时注册到一组controller里面，并且同时注册到同一组namesrv中
![image](https://github.com/apache/rocketmq/assets/57432151/6f4a413a-e5ab-4eff-90e4-ade8e35414fd)

**Comment by figuewang** [comment]
主从同步有性能损耗，从节点越多性能肯定更差
